src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -36,6 +35,66 @@ public class NodeSettings
         /// <summary>Version of the protocol the current implementation supports.</summary>
         public const ProtocolVersion SupportedProtocolVersion = ProtocolVersion.SENDHEADERS_VERSION;
 
+        /// <summary>Factory to create instance logger.</summary>
+        public ILoggerFactory LoggerFactory { get; private set; }
+
+        /// <summary>Instance logger.</summary>
+        public ILogger Logger { get; private set; }
+
+        /// <summary>Configuration related to logging.</summary>
+        public LogSettings Log { get; private set; }
+
+        /// <summary>List of paths to important files and folders.</summary>
+        public DataFolder DataFolder { get; private set; }
+
+        /// <summary>Path to the data directory. This value is read-only and is set in the constructor's args.</summary>
+        public string DataDir { get; private set; }
+
+        /// <summary>Path to the configuration file. This value is read-only and is set in the constructor's args.</summary>
+        public string ConfigurationFile { get; private set; }
+
+        /// <summary>Combined command line arguments and configuration file settings.</summary>
+        public TextFileConfiguration ConfigReader { get; private set; }
+
+        /// <summary>Option to skip (most) non-standard transaction checks, for testnet/regtest only.</summary>
+        public bool RequireStandard { get; set; }
+
+        /// <summary>Supported protocol version.</summary>
+        public ProtocolVersion ProtocolVersion { get; private set; }
+
+        /// <summary>Specification of the network the node runs on - regtest/testnet/mainnet.</summary>
+        public Network Network { get; private set; }
+
+        /// <summary>The node's user agent.</summary>
+        public string Agent { get; private set; }
+
+        /// <summary>The node's user agent prefix.</summary>
+        public string AgentPrefix { get; private set; }
+
+        /// <summary>The node's user agent (with optional prefix) that will be shared with peers in the version handshake.</summary>
+        public string AgentWithPrefix
```

Comment:
Now we have Agent, AgentPrefix and AgentwithPrefix.
I think if the user specifies an -agentprefix, the agent invludes this prefix and we're only interested in the Agent.
We shoulld only have 1 property for that really.

Reply:
Ok. I will follow and alternative approach and change the legacy code from:

`this.Agent = string.IsNullOrEmpty(agentPrefix) ? this.Agent : $"{agentPrefix}-{this.Agent}"; `

to:
```
var agent = this.ConfigReader.GetOrDefault("agent", "");
this.Agent = string.IsNullOrEmpty(agentPrefix) ? agent : $"{agentPrefix}-{agent}"; 
```

so that if multiple calls to LoadConfiguration are ever performed it won't mess with the agent value.

Reply:
Done.